applications/RomApplication/python_scripts/petrov_galerkin_training_utility.py

```diff
@@ -95,17 +95,12 @@ def __GetPetrovGalerkinTrainingDefaultSettings(cls):
         }""")
         return default_settings
 
-    def __CalculateResidualBasis(self):
-        # Set up the snapshots matrix for new basis
-        n_steps = len(self.time_step_snapshots_matrix_container)
-        snapshots_matrix = self.time_step_snapshots_matrix_container[0]
-        for i in range(1,n_steps):
-            del self.time_step_snapshots_matrix_container[0] # Avoid having two matrices, numpy does not concatenate references.
-            snapshots_matrix = np.c_[snapshots_matrix,self.time_step_snapshots_matrix_container[0]]
+    def __CalculateResidualBasis(self, snapshots_matrix):
+        if snapshots_matrix is None:
+            snapshots_matrix = self._GetSnapshotsMatrix()
         u_left,s_left,_,_ = RandomizedSingularValueDecomposition(COMPUTE_V=False).Calculate(
             snapshots_matrix,
             self.svd_truncation_tolerance)
-        del(snapshots_matrix)
```

Comment:
Why are you keeping this ``snapshots_matrix``? This could cause you to run out of memory.